src/main/java/com/minecolonies/coremod/client/gui/WindowBuildTool.java

```diff
@@ -274,273 +417,222 @@ public void onClosed()
-
-        changeSchematic();
+        schematicsDropDownList.selectNext();
     }
 
-    /*
-     * ---------------- Button Handling -----------------
+    /**
+     * Go to the previous schematic.
      */
+    private void previousSchematic()
+    {
+        schematicsDropDownList.selectPrevious();
+    }
 
     /**
-     * Get all styles from the folders.
-     *
-     * @return list of style strings.
+     * Update the list a available schematics.
      */
-    private List<String> getStyles()
+    private void updateSchematics()
     {
-        try
+        String schematic = "";
+        if (schematicsDropDownList.getSelectedIndex() > -1 && schematicsDropDownList.getSelectedIndex() < schematics.size())
         {
-            Structures.loadStyleMaps(new File(Minecraft.getMinecraft().mcDataDir, "minecolonies/decorations").toPath());
+            schematic = schematics.get(schematicsDropDownList.getSelectedIndex());
         }
-        catch (final IOException e)
+        final String currentSchematic = schematic.isEmpty() ? "" : (new Structures.StructureName(schematic)).getSchematic();
+        String section = sections.get(sectionsDropDownList.getSelectedIndex());
+        String style = styles.get(stylesDropDownList.getSelectedIndex());
+        schematics = Structures.getSchematicsFor(section, style);
+        int newIndex = -1;
+        for (int i = 0; i < schematics.size(); i++)
         {
-            Log.getLogger().warn("No additional files found", e);
+            Structures.StructureName sn = new Structures.StructureName(schematics.get(i));
+            if (sn.getSchematic().equals(currentSchematic))
+            {
+                newIndex = i;
+                break;
+            }
         }
 
-        if (hutDec.isEmpty())
+        if (newIndex == -1)
         {
-            return Collections.emptyList();
+            newIndex = 0;
         }
 
-        if (Settings.instance.isInHutMode())
-        {
-            return Structures.getStylesForHut(hutDec.get(hutDecIndex));
-        }
-        else
-        {
-            return Structures.getStylesForDecoration(hutDec.get(hutDecIndex));
-        }
+        final boolean enabled = schematics.size() > 1;
+        findPaneOfTypeByID(BUTTON_PREVIOUS_SCHEMATIC_ID, Button.class).setEnabled(enabled);
+        findPaneOfTypeByID(DROPDOWN_SCHEMATIC_ID, DropDownList.class).setEnabled(enabled);
+        findPaneOfTypeByID(BUTTON_NEXT_SCHEMATIC_ID, Button.class).setEnabled(enabled);
+        schematicsDropDownList.setSelectedIndex(newIndex);
     }
 
     /**
-     * Changes the current structure.
-     * Set to button position at that time
+     * called every time one of the dropdownlist changed.
+     *
+     * @param list the dropdown list which change
+     * @param index is the index selected in the list
      */
-    private void changeSchematic()
+    private void onDropDownListChanged(final DropDownList list, final int index)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "index". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1172)